002-routing-params-views/src/app/common/app.component.js

```diff
@@ -0,0 +1,24 @@
+(function(){
+    'use strict';
+
+    var app = {
+        templateUrl: 'app/common/app.html',
+        controller: 'AppController',
+        controllerAs: 'appCtrl'
+    };
+
+    angular
+        .module('common')
+        .component('app', app)
+        .config(appCofig);
+
+    /** @ngInject */
+    function appCofig($stateProvider) {
+        $stateProvider
+            .state('app', {
+                redirectTo: 'dashboard',
+                url: '/app',
+                template: '<app layout="row" flex></app>'
+            })
+    }
+})();
```

Comment:
I Would put in different files the component definition and the route, if you do it like this then the objective of the  component concept is not being achieved